File: shared/api_contracts/user.contract.ts
export interface UserContract {
  id: string;
  email: string;
  firstName?: string;
  lastName?: string;
  role: "player" | "coach" | "admin";
}

export interface AuthResponse {
  success: boolean;
  data: {
    user: UserContract;
    accessToken: string;
  };
  error: string | null;
}

File: frontend_flutter/lib/shared/models/user_contract.dart
class UserContract {
  final String id;
  final String email;
  final String? firstName;
  final String? lastName;
  final String role;

  UserContract({
    required this.id,
    required this.email,
    this.firstName,
    this.lastName,
    required this.role,
  });

  factory UserContract.fromJson(Map<String, dynamic> json) {
    return UserContract(
      id: json['id'],
      email: json['email'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      role: json['role'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'email': email,
      'firstName': firstName,
      'lastName': lastName,
      'role': role,
    };
  }
}
